src/declarations.d.ts

```diff
@@ -1 +1 @@
-declare module 'react-merge-refs'
+// declare module 'react-merge-refs' // removed to fix typings issues with primitives refs loosing types. react-merge-refs is written in typescript, and has it's own types.
```

Comment:
this was preventing 'react-merge-refs' from infer the types in primitive jsx ref prop